File: Advanced Methods of Programming/JavaClientServer/Model/src/main/java/Model/Inscriere.java
package Model;

import java.io.Serializable;

public class Inscriere implements HasId<Integer>, Serializable {
    private int idInscriere;
    private int idProba;
    private int idParticipant;


    public Inscriere(int idProba, int idParticipant) {
        this.idInscriere = Integer.parseInt(String.valueOf(idProba)+String.valueOf(idParticipant));
        this.idProba = idProba;
        this.idParticipant = idParticipant;
    }

    public Integer getId() { return idInscriere; }

    public void setId(Integer idInscriere) { this.idInscriere = idInscriere; }

    public int getIdProba() { return idProba; }

    public void setIdProba(int idProba) { this.idProba = idProba; }

    public int getIdParticipant() { return idParticipant; }

    public void setIdParticipant(int idParticipant) { this.idParticipant = idParticipant; }
}

File: Advanced Methods of Programming/Lab1C-MPP/Lab1C-MPP/Domain/Inscriere.cs
namespace Lab1C_MPP.Domain
{
    class Inscriere:IHasID<int>
    {
        private int IdInscriere;
        private int IdProba;
        private int IdParticipant;

        public Inscriere(int IdPr, int IdPa)
        {
            this.IdInscriere = int.Parse(IdPr.ToString()+IdPa.ToString());
            this.IdProba = IdPr;
            this.IdParticipant = IdPa;
        }

        public int GetId() { return this.IdInscriere; }

        public int GetIdProba() { return this.IdProba; }

        public int GetIdParticipant() { return this.IdParticipant; }
    }
}
